visualization/src/point_picking_event.cpp

```diff
@@ -239,13 +227,17 @@ pcl::visualization::PointPickingCallback::performAreaPick (vtkRenderWindowIntera
 
       std::vector<int> actorIndices;
       assert (GlobalIDs->GetSize());
+      
       actorIndices.reserve (selected->GetNumberOfPoints ());
       for (vtkIdType i = 0; i < selected->GetNumberOfPoints (); i++)
         actorIndices.push_back(static_cast<int>(GlobalIDs->GetValue(i)));
         
       pt_numb+= selected->GetNumberOfPoints ();
+      
+      const auto actorSelected = std::find_if(cam_ptr->cbegin(), cam_ptr->cend(), [&actor](const auto& cloud_actor) { return cloud_actor.second.actor == actor; });
+      const std::string name = (actor != cam_ptr->cend()) ? actor->first : "not_found";
       indices.push_back(actorIndices);
-      actors->AddItem(actor);
+      cloudIndices.insert({name, std::move(indices)});
```

Comment:
There's no `indices` so you can remove `indices` and replace `actorIndices` by a reference using:
* Push the code to find the name (L237-238) to L227
* Replace 228 with `auto& actorIndices = cloudIndices[name];`
* Delete L2390-240

Reply:
I made clean up so now should be ok.